doc/source/index.rst

```diff
@@ -9,84 +9,179 @@ Ray
+        return x * x
+
+    futures = [f.remote(i) for i in range(4)]
+    print(ray.get(futures))
+
+To use Ray's actor model:
+
+.. code-block:: python
+
+    ray.init()
+
+    @ray.remote
+    class Counter():
+        def __init__(self):
+            self.n = 0
+
+        def inc(self):
+            self.n += 1
+
+        def read(self):
+            return self.n
+
+    counters = [Counter.remote() for i in range(4)]
+    [c.increment.remote() for c in counters]
+    futures = [c.read.remote() for c in counters]
+    print(ray.get(futures))
+
+
+Ray programs can run on a single machine, and can also seamlessly scale to large clusters. To execute the above Ray script in the cloud, just download `this configuration file <https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml>`__, and run:
+
+``ray submit [CLUSTER.YAML] example.py --start``
+
+See more details in the `Cluster Launch page <autoscaling.html>`_.
+
+Tune Quick Start
+----------------
+
+`Tune`_ is a scalable framework for hyperparameter search with a focus on deep learning and deep reinforcement learning.
+
+.. code-block:: python
+
+   import torch.optim as optim
+   from ray import tune
+   from ray.tune.examples.mnist_pytorch import get_data_loaders, Net, train, test
+
+
+   def train_mnist(config):
+       train_loader, test_loader = get_data_loaders()
+       model = Net(config)
+       optimizer = optim.SGD(model.parameters(), lr=config["lr"])
+       for i in range(10):
+           train(model, optimizer, train_loader)
+           acc = test(model, test_loader)
+           tune.track.log(mean_accuracy=acc)
+
+
+   analysis = tune.run(
+       train_mnist,
+       stop={"mean_accuracy": 0.98},
+       config={"lr": tune.grid_search([0.001, 0.01, 0.1])})
+
+   print("Best config: ", analysis.get_best_config())
 
 .. _`Tune`: tune.html
+
+RLlib Quick Start
+-----------------
+
+`RLlib`_ is an open-source library for reinforcement learning that offers both high scalability and a unified API for a variety of applications.
+
+.. code-block:: bash
+
+  pip install tensorflow  # or tensorflow-gpu
+  pip install ray[rllib]  # also recommended: ray[debug]
+
+.. code-block:: python
+
+    import gym
+    from gym.spaces import Discrete, Box
+    from ray import tune
+
+    class SimpleCorridor(gym.Env):
+        def __init__(self, config):
+            self.end_pos = config["corridor_length"]
+            self.cur_pos = 0
+            self.action_space = Discrete(2)
+            self.observation_space = Box(0.0, self.end_pos, shape=(1, ))
+
+        def reset(self):
+            self.cur_pos = 0
+            return [self.cur_pos]
+
+        def step(self, action):
+            if action == 0 and self.cur_pos > 0:
+                self.cur_pos -= 1
+            elif action == 1:
+                self.cur_pos += 1
+            done = self.cur_pos >= self.end_pos
+            return [self.cur_pos], 1 if done else 0, done, {}
+
+    tune.run(
```

Comment:
This example takes too long to run, can we make it finish in under 30 seconds?

Reply:
I think this is expected for RL.